cmd/snap/cmd_paths.go

```diff
@@ -0,0 +1,65 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2018 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package main
+
+import (
+	"fmt"
+
+	"github.com/jessevdk/go-flags"
+
+	"github.com/snapcore/snapd/dirs"
+	"github.com/snapcore/snapd/i18n"
+)
+
+var pathsHelp = i18n.G("Print system paths")
+var longPathsHelp = i18n.G(`
+The paths command prints the list of paths detected and used by snapd.
+`)
+
+type cmdPaths struct{}
+
+func init() {
+	addDebugCommand("paths", pathsHelp, longPathsHelp, func() flags.Commander {
+		return &cmdPaths{}
+	}, nil, nil)
+}
+
+func (cmd cmdPaths) Execute(args []string) error {
+	if len(args) > 0 {
+		return ErrExtraArgs
+	}
+
+	w := tabWriter()
+	defer w.Flush()
+
+	// TODO: include paths reported by snap-confine
+	for _, p := range []struct {
+		name string
+		path string
+	}{
+		{"snap-mount-dir", dirs.SnapMountDir},
+		{"snap-bin-dir", dirs.SnapBinariesDir},
+		{"distro-libexec", dirs.DistroLibExecDir},
+	} {
+		fmt.Fprintf(w, "%s:\t%s\n", p.name, p.path)
```

Comment:
A format like this might be better here:
```
SNAPD_MOUNT=...
SNAPD_BIN=...
SNAPD_LIBEXEC=...
```
This allows usage in shell for analysis, and it also uses the $SNAPD_ prefix, which is what we generally have for snapd-related content. We usually reserve the $SNAP_ prefix for things which are for a snap, with $SNAP being the path to the snap directory under /snap. We also don't use _DIR for $SNAP_*, so we don't need it here either.